packages/web-components/src/components/masthead/masthead-l1.ts

```diff
@@ -365,24 +371,44 @@ class C4DMastheadL1 extends HostListenerMixin(StableSelectorMixin(LitElement)) {
       return markup;
     };
 
-    if (cta && cta?.title) {
-      if (cta?.ctaType === CTA_TYPE.CHAT) {
+    /**
+     * Maps CTA_TYPE values to CMApp CTA options.
+     *
+     * @see https://github.ibm.com/live-advisor/cm-app/blob/master/docs/cm-doc.md#calls-to-action
+     */
+    const iconMap = new Map([
+      [L1_CTA_TYPES.CONTACT_US, Chat16()],
+      [L1_CTA_TYPES.CHAT_NOW, Chat16()],
+      [L1_CTA_TYPES.EMAIL_US, Email16()],
+      [L1_CTA_TYPES.CALL_US, Phone16()],
+      [L1_CTA_TYPES.BOOK_A_CONSULTATION, Calendar16()],
+      [L1_CTA_TYPES.REQUEST_A_DEMO, Demo16()],
+      [L1_CTA_TYPES.REQUEST_A_QUOTE, Quote16()],
+    ]);
+
+    if (cta) {
+      const { ctaType } = cta;
+      if (ctaType && iconMap.has(ctaType as L1_CTA_TYPES)) {
+        const icon = iconMap.get(ctaType as L1_CTA_TYPES);
         return html`
           <button
             class="${classname}"
-            data-ibm-contact="contact-link"
+            data-ibm-contact="${ctaType}-link"
             aria-label="${ifDefined(contactCtaLabel)}">
             ${desktopWrapper(
-              html`<span data-ibm-contact="contact-text">${cta.title}</span
-                >${Chat16()}`
+              html`<span data-ibm-contact="${ctaType}-text"
+                  >${cta?.title ?? ''}</span
+                >${icon}`
             )}
           </button>
         `;
-      } else if (cta?.url) {
+      } else if (cta?.title && cta?.url) {
         const icon = isMobileVersion ? ArrowRight16() : '';
         return html`
           <a class="${classname}" href="${cta.url}">
-            ${desktopWrapper(html`${cta.title}${icon}`)}
+            ${desktopWrapper(html`
+              <slot name="cta-text">${cta.title}</slot>${icon}
```

Comment:
Two questions, how come we're introducing a `<slot>` here? Is this like a little bonus feature to allow overriding the text from the markup?  Second, why does tha above conditional not get a `<slot name="cta-text"></slot>` but here it does?